packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2043,8 +2126,42 @@ abstract class BuildContext {
   /// );
   /// ```
   /// {@end-tool}
+  @Deprecated('Use findAncestorStateOfType instead')
   State ancestorStateOfType(TypeMatcher matcher);
```

Comment:
We should probably mark TypeMatcher as deprecated as well since that will go away as soon as we delete the old deprecated API?

Reply:
Done.